assets/js/editor-components/products-control/index.js

```diff
@@ -66,7 +66,13 @@ const ProductsControl = ( {
 	return (
 		<SearchListControl
 			className="woocommerce-products"
-			list={ products }
+			list={ products.map( ( product ) => {
+				const formattedSku = product.sku ? '(' + product.sku + ')' : '';
+				return {
+					...product,
+					name: `${ product.name } ${ formattedSku }`,
```

Comment:
Very tiny nit, but, wouldn't there be a space at the end of the name if the product doesn't have an SKU?

```suggestion
				const formattedSku = product.sku ? ' (' + product.sku + ')' : '';
				return {
					...product,
					name: `${ product.name }${ formattedSku }`,
```